src/docfx/lib/ProcessUtility.cs

```diff
@@ -160,25 +160,34 @@ public static async Task RunInsideMutex(string mutexName, Func<Task> action)
             Debug.Assert(!string.IsNullOrEmpty(mutexName));
 
             // avoid the RunInsideMutex to be nested used
+            // doesn't support to require a lock before releasing a lock
+            // which will cause deadlock
             Debug.Assert(t_innerCall.Value == null);
```

Comment:
throw? Debug.Assert only runs in DEBUG mode, test will fail in release mode.

Reply:
zan